internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,604 @@
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      databaseSecretName,
+			Namespace: r.Namespace,
+		},
+	}
+
+	// pass, err := generatePassword()
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// TODO(djzager): Use legit password
+	secretData := map[string]string{
+		"db.host":     "localhost",
+		"db.user":     "admin",
+		"db.password": "abcdefg",
+		"db.name":     "installer",
+		"db.port":     strconv.Itoa(int(databasePort)),
+	}
+
+	secretType := corev1.SecretTypeOpaque
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, secret, r.Scheme); err != nil {
+			return err
+		}
+		secret.StringData = secretData
+		secret.Type = secretType
+		return nil
+	}
+
+	return secret, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *adiiov1alpha1.AgentServiceConfig) (*appsv1.Deployment, controllerutil.MutateFn) {
+	ocmSSOSecretName := "assisted-installer-sso"
+	s3SecretName := "assisted-installer-s3"
+	publicS3SecretName := "assisted-installer-public-s3"
+	assistedServiceConfigMapName := "assisted-service-config"
+	maxUnavailable := intstr.FromString("50%")
+	maxSurge := intstr.FromString("100%")
+
+	serviceContainer := corev1.Container{
+		Name:  assistedServiceContainerName,
+		Image: ServiceImage(),
+		Ports: []corev1.ContainerPort{
+			{
+				ContainerPort: servicePort,
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Env: []corev1.EnvVar{
+			// TODO(djzager): clean up unneccessary environment variables
+			newSecretEnvVar("OCM_SERVICE_CLIENT_ID", "ocm-service.clientId", ocmSSOSecretName),
+			newSecretEnvVar("OCM_SERVICE_CLIENT_SECRET", "ocm-service.clientSecret", ocmSSOSecretName),
+			newSecretEnvVar("AWS_SECRET_ACCESS_KEY", "aws_secret_access_key", s3SecretName),
+			newSecretEnvVar("AWS_ACCESS_KEY_ID", "aws_access_key_id", s3SecretName),
+			newSecretEnvVar("S3_REGION", "aws_region", s3SecretName),
+			newSecretEnvVar("S3_BUCKET", "bucket", s3SecretName),
+			newSecretEnvVar("S3_ENDPOINT_URL", "endpoint", s3SecretName),
+			newSecretEnvVar("AWS_SECRET_ACCESS_KEY_PUBLIC", "aws_secret_access_key", publicS3SecretName),
+			newSecretEnvVar("AWS_ACCESS_KEY_ID_PUBLIC", "aws_access_key_id", publicS3SecretName),
+			newSecretEnvVar("S3_REGION_PUBLIC", "aws_region", publicS3SecretName),
+			newSecretEnvVar("S3_BUCKET_PUBLIC", "bucket", publicS3SecretName),
+			newSecretEnvVar("S3_ENDPOINT_URL_PUBLIC", "endpoint", publicS3SecretName),
+
+			// TODO: Should these be in a ConfigMap? the assisted-service-config CM?
+			newEnvVar("ISO_IMAGE_TYPE", "minimal-iso"),
+			newEnvVar("S3_USE_SSL", "false"),
+			newEnvVar("LOG_LEVEL", "info"),
+			newEnvVar("LOG_FORMAT", "text"),
+			newEnvVar("INSTALL_RH_CA", "false"),
+			newEnvVar("REGISTRY_CREDS", ""),
+			newEnvVar("AWS_SHARED_CREDENTIALS_FILE", "/etc/.aws/credentials"),
+			newEnvVar("DEPLOY_TARGET", "ocp"),
```

Comment:
should probably be `k8s` we saw that OCP is not working with none bare metal deployment.